api/v1alpha8/types.go

```diff
@@ -87,6 +87,12 @@ type RouterFilter struct {
 	NotTagsAny  string `json:"notTagsAny,omitempty"`
 }
 
+type SubnetSpec struct {
+	// +required
```

Comment:
could we please do better on the doc strings for each of the fields, thanks!

Reply:
Done. I just realized that in the future we might want to add a `SubnetPool` field here and that would make `CIDR` not required anymore (can be substituted by the `SubnetPool`). Will we be able to drop a constraint like that? @mdbooth?

Reply:
I think we'll have to make these fields mutually exclusive but I don't think it's a problem today.